src/openjd/adaptor_runtime/process/_logging_subprocess.py

```diff
@@ -64,7 +69,27 @@ def __init__(
             # In Windows, this is required for signal. SIGBREAK will be sent to the entire process group.
             # Without this one, current process will also get the SIGBREAK and may react incorrectly.
             popen_params.update(creationflags=subprocess.CREATE_NEW_PROCESS_GROUP)  # type: ignore[attr-defined]
-        self._process = subprocess.Popen(**popen_params)
+
+        try:
+            self._process = subprocess.Popen(**popen_params)
+
+        except FileNotFoundError as fnf_error:
+            # In ManagedProcess we prepend the executable to the list of arguments before creating a LoggingSubprocess
+            executable = args[0]
+            exe_path = shutil.which(executable)
+
+            # If we didn't find the executable found by which
+            if exe_path is not None:
+                raise FileNotFoundError(
+                    f"Could not find adaptor executable at: {exe_path} using alias {executable}\n"
+                    f"Error:{fnf_error}"
+                )
+
+            raise FileNotFoundError(
+                f"Could not find the executable associated with the adaptor: {executable}\n"
+                f"Is the executable on the PATH or in the startup directory?\n"
```

Comment:
In this context, there isn't necessarily an adaptor, so we should make the message more general

```suggestion
                f"Could not find the specified command: {executable}\n"
                f"Ensure the command is available from the PATH environment variable.\n"
```